### ISSUE TITLE
sway autostart apps are difficult/impossible to override

### ISSUE THREAD

**Issue by tve** [manjaro-sway/manjaro-sway#356]
Some of the stuff done in `/etc/sway/config.d/99-autostart.conf` is difficult or impossible to override/customize. Specifically, most of the commands can only be disabled by uninstalling packages, which is not so cool if I want to have the commands around to experiment with. The starting of `foot --server` is something that cannot be disabled. I don't want the server mode but I do use foot, so I'm resorting to a `pkill foot` in my local autostart, whcih is rather hacky.

I believe it would be cleaner and more in line with other parts of the sway config if a definitions file had something like `set $autotiling_cmd=...` and then the conf file laucnhed that command if it's not empty.

**Comment by boredland** [comment]
Cool, but probably challenging idea! I am very open to a specific proposal though.

**Comment by negativeExponent** [comment]
if you want to mess around with 99-autostart-applications.conf, you may want to get rid of the "dex" command as well, and just add any of them manually individully if you them to start (/etc/xdg/autostart, ~/.config/autostart)

check also user services started with systemd, foot server is one of them and not sure if manjaro enables it by default (not a manjaro user)

**Comment by tve** [comment]
> I am very open to a specific proposal though.

The difficulty seems to be that you need two checks: does the executable exist and does the user want to run it. If I ignore the first one and focus on the launching of foot server an an example, I'd say:

- in the definitions file define `set $foot_server /usr/share/sway/scripts/foot.sh --server`
- in the autostart file use `[ -n "$foot_server" ] && $foot_server`

This is very similar to the way various forms of $term* are defined and then launched. If I don't want to run foot I should be able to simply put `set $foot_server` in my local definitions. If I want to launch it differently, I can set the variable to a slightly different commandline.

To add the check for `-x $(command -v foot)` one option would be to use something like:
```
set $foot_server [ -x "$(command -v foot)" ] && /usr/share/sway/scripts/foot.sh --server
```
Basically this moves the command from autostart-applications.conf to definitions.

An alternative, which might be less "flexible" but simpler might be to prefix all commands in autostart-applications.conf with a `-n $do_autostart`, e.g.:
```
    '[ -n $do_autostart -a -x "$(command -v foot)" ] && /usr/share/sway/scripts/foot.sh --server'
```
The definitions would set do_autostart=1 but I can turn the whole shebang off by setting it to empty string in my definitions, then copy the autostart lines I want with any modifications I want to my own autostart.
```

**Comment by boredland** [comment]
voila: https://github.com/manjaro-sway/manjaro-sway/blob/main/SUPPORT.md#how-can-i-remove-an-auto-starting-application

**Comment by boredland** [comment]
will make a release soon.
